lib/alchemy/configurations/auth.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module Alchemy
+  module Configurations
+    class Auth < Alchemy::Configuration
```

Comment:
I don't think we should add another configuration class just for auth. The methods have conflict free names and it only makes it harder to read and use. Alchemy.config.user_class is perfect. Alchemy.config.auth.user_class is to verbose and does not provide anything useful.